### PR TITLE
Task5/jdbc: database with 3 different tables

### PR DESCRIPTION
**Pull request by qnislavnachev**
there is a database with 3 different tables and 1 manager class that
register into database and show tables

### CODE REVIEW

**Comment by qnislavnachev** [comment]
this task is ready, need a review

File: src/task5/informator/Manager.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        return getAdaptedList(query, Objects.CONTACT);
+    }
+
+    @Override
+    public List<Address> findAllAddresses() {
+        String query = "select * from Addresses";
+        return getAdaptedList(query, Objects.ADDRESS);
+    }
+
+    private List getAdaptedList(String query, Enum e) {
```
Comment (tsony-tsonev) [inline src/task5/informator/Manager.java]: in this method you did not remove the duplication you just wrote it in one method, if we want to add one more table we should add more `if` clauses

```java
public interface ResultSetAdapter<T> {
    T adapt(ResultSet resultSet) throws SQLException;
}

public <T> List<T> listAdapter(Connection connection, String query, ResultSetAdapter<T> adapter){
    ResultSet rs;
    List<T> list = new ArrayList<T>();
    try {
      rs = connection.createStatement().executeQuery(query);
      while (rs.next()) {
        list.add(adapter.adapt(rs));
      }
    } catch (SQLException e) {
      e.printStackTrace();
    }

    return list;
}

@org.junit.Test
public void adaptPersonsResultSet() throws Exception {
    Connection connection = null;
    List<Person> persons = listAdapter(connection, "select * from persons", new ResultSetAdapter<Person>() {
      @Override
      public Person adapt(ResultSet resultSet) throws SQLException {
        return new Person(resultSet.getInt("id"), resultSet.getString("name"), resultSet.getInt("age"));
      }
    });
}
```

look at this implementation, with this approach you should only call the listAdapter the same way it is called in the test  inside your find methods (only the adapted object will be different) and then you can just return the adapted list, and you can do it for as many tables as you wish

File: src/task5/informator/Database.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package task5.informator;
+
+import task5.core.Address;
+import task5.core.Contact;
+import task5.core.User;
+
+import java.util.List;
+
+public interface Database {
```
Comment (tsony-tsonev) [inline src/task5/informator/Database.java]: this should be splint into 3 interfaces one for each type of repository, you can leave the implementation just one soo you can see the use of the listAdapter but be sure that you implement all of the split interfaces

File: src/task5/informator/Manager.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                }
+                return list;
+            }
+        } catch (SQLException e1) {
+            e1.printStackTrace();
+        }
+        return null;
+    }
+
+    private void insertUser(String query, User user) {
```
Comment (tsony-tsonev) [inline src/task5/informator/Manager.java]: all these insert methods are used just once, soo I'm not sure they are needed as separate methods, but you can leave them as they are for better readability.

File: src/task5/informator/Database.java (hunk 2 above)
Comment (qnislavnachev) [inline src/task5/informator/Database.java]: i think now its fixed

**Comment by tsony-tsonev** [comment]
Good but do not write answers in the commit messages, also you still have to split the database  interface into repository interfaces for each table, also try to use the columns names instead of they integer id when working with the result set.

**Comment by qnislavnachev** [comment]
splitting interface is done now i have interface for each table

**Comment by tsony-tsonev** [comment]
LGTM

**Pull request merged**